modules/services/gitlab-runner.nix

```diff
@@ -0,0 +1,643 @@
+              Runner-specific command script executed after code is pulled,
+              just before build executes.
+            '';
+          };
+          postBuildScript = mkOption {
+            type = types.nullOr types.path;
+            default = null;
+            description = ''
+              Runner-specific command script executed after code is pulled
+              and just after build executes.
+            '';
+          };
+          tagList = mkOption {
+            type = types.listOf types.str;
+            default = [ ];
+            description = ''
+              Tag list.
+            '';
+          };
+          runUntagged = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              Register to run untagged builds; defaults to
+              <literal>true</literal> when <option>tagList</option> is empty.
+            '';
+          };
+          limit = mkOption {
+            type = types.int;
+            default = 0;
+            description = ''
+              Limit how many jobs can be handled concurrently by this service.
+              0 (default) simply means don't limit.
+            '';
+          };
+          requestConcurrency = mkOption {
+            type = types.int;
+            default = 0;
+            description = ''
+              Limit number of concurrent requests for new jobs from GitLab.
+            '';
+          };
+          maximumTimeout = mkOption {
+            type = types.int;
+            default = 0;
+            description = ''
+              What is the maximum timeout (in seconds) that will be set for
+              job when using this Runner. 0 (default) simply means don't limit.
+            '';
+          };
+          protected = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              When set to true Runner will only run on pipelines
+              triggered on protected branches.
+            '';
+          };
+          debugTraceDisabled = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              When set to true Runner will disable the possibility of
+              using the <literal>CI_DEBUG_TRACE</literal> feature.
+            '';
+          };
+        };
+      });
+    };
+  };
+  config = mkIf cfg.enable {
+
+    users.users.gitlab-runner =
+      { name = "gitlab-runner";
+        uid = mkDefault 532;
+        gid = mkDefault config.users.groups.gitlab-runner.gid;
+        home = mkDefault "/var/lib/gitlab-runner";
+        shell = "/bin/bash";
+        description = "Gitlab agent user";
+      };
+    users.groups.gitlab-runner =
+      { name = "gitlab-runner";
+        gid = mkDefault 532;
+        description = "Gitlab agent user group";
+      };
+
+
+    system.activationScripts.preActivation.text = let user = config.users.users.gitlab-runner; in ''
```

Comment:
The launchd module should probably handle this automatically but the `preActivation`, etc. are intended purely for users. Using priorities in a configuration would break the module which isn't ideal.